src/main/java/com/microsoft/azure/vmagent/AzureVMManagementServiceDelegate.java

```diff
@@ -749,6 +778,59 @@ public static AzureVMAgent parseResponse(
         return sizes;
     }
 
+    private static Map<String, String> getDefaultImagePublisher() {
```

Comment:
It may be better to put all properties of one image into an object (Map or a custom object) and have a helper method to return the object for an image name, may be easier to read and maintain in future.